.github/workflows/ubuntu.yml

```diff
@@ -156,17 +154,17 @@ jobs:
       - name: archive-files
         # We only run tests on the version compiled with gcc, so we
         # only need to archive that one.
-        if: ${{ matrix.compiler-version }} == gcc
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
         run: |
           cd ../
-          tar cfz ${{ matrix.compiler-version }}.tar.gz downward lib
+          tar cfz ${{ matrix.compiler-version.cc }}.tar.gz downward lib
 
       - name: upload-files
-        if: ${{ matrix.compiler-version }} == gcc
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
         uses: actions/upload-artifact@v1
         with:
           name: compiled-planner-${{ matrix.ubuntu-version }}
-          path: /home/runner/work/downward/${{ matrix.compiler-version }}.tar.gz
+          path: /home/runner/work/downward/${{ matrix.compiler-version.cc }}.tar.gz
 
 
   test:
```

Comment:
Instead of the if/else block at the end of the file, you can use "tox -e py,..." and tox will use the available Python version.